src/main.js

```diff
@@ -1,33 +1,67 @@
-//Create variables targetting the relevant DOM elements here 👇
 
 var coverImage = document.querySelector(".cover-image");
 var title = document.querySelector(".cover-title");
 var tagLine = document.querySelector(".tagline");
 var descriptor1 = document.querySelector(".tagline-1");
 var descriptor2 = document.querySelector(".tagline-2");
-var randomCoverButton = document.querySelector(".random-cover-button");
-
+var btnRandomCover = document.querySelector(".random-cover-button");
+var btnMakeNewCover = document.querySelector(".make-new-button");
+var btnViewSavedCover = document.querySelector(".view-saved-button");
+var btnHome = document.querySelector(".home-button");
+var btnSaveCover = document.querySelector(".save-cover-button");
+var homeView = document.querySelector(".home-view");
+var savedView = document.querySelector(".saved-view");
+var formView = document.querySelector(".form-view");
 
 
 var savedCovers = [
   new Cover("http://3.bp.blogspot.com/-iE4p9grvfpQ/VSfZT0vH2UI/AAAAAAAANq8/wwQZssi-V5g/s1600/Do%2BNot%2BForsake%2BMe%2B-%2BImage.jpg", "Sunsets and Sorrows", "sunsets", "sorrows")
 ];
 
+window.addEventListener("load", createRandomCover);
```

Comment:
could be good to query all your elements and bind all your click handlers after load. could be something like window.addEventListener("load", initialize). Then
`function initialize() {
  query all dom elements
  add all event listeners
  createRandomCover()
}
`

Maybe the bug you were running into is a race condition where sometimes the page was loaded by the time you queried "Make new cover" button, and sometimes the JS executed before the button had been added to the DOM.